### ISSUE TITLE
Allow for a logger to be attached to client

### ISSUE THREAD

**Issue by leh** [ignacio-chiazzo/ruby_whatsapp_sdk#121]
I am exploring the whatapp api with this great gem.

As I'm new to the whatsapp api, I do a lot of stupid mistakes. I have a hard time debbugging the mistakes I do because there no way to attach a logger to Faraday to see the HTTP requests (with headers and bodies).

It would be nice if while configuring the client one could set a logger:
```ruby
# config/initializers/whatsapp_sdk.rb
WhatsappSdk.configure do |config|
  config.access_token = ACCESS_TOKEN
  config.logger = Rails.logger
end
```

**Comment by ignacio-chiazzo** [comment]
Thanks for opening an issue. I like that. 

I also think we should print the errors by default even if the logger isn't configured.

**Comment by ignacio-chiazzo** [comment]
Hi @chahmedejaz sure! Feel free to take it

**Comment by ignacio-chiazzo** [comment]
Hint: you will need to update the Faraday connection to pass a Middelware for logging. The code lives here https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/blob/2373a725a64bfdc50966f38b7a8cd8b7e3587019/lib/whatsapp_sdk/api/client.rb#L73-L80

**Comment by chahmedejaz** [comment]
Hey @ignacio-chiazzo - really sorry, I'm working on it. submitting the PR soon. Thanks.

**Comment by ignacio-chiazzo** [comment]
@chahmedejaz thanks! If you want to chat/pair I'm available. Feel free to message me on [X](https://twitter.com/ignaciochiazzo)

**Comment by chahmedejaz** [comment]
> @chahmedejaz thanks! If you want to chat/pair I'm available. Feel free to message me on [X](https://twitter.com/ignaciochiazzo)

Sure, I'll. 😊
I've implemented the fix, working its specs 😊. Thanks

**Comment by chahmedejaz** [comment]
Hey @ignacio-chiazzo - The PR is ready for review. Would love to hear your thoughts on it. Thanks.

**Comment by chahmedejaz** [comment]
@ignacio-chiazzo - I was wondering if we have any work left for this issue. Thanks.

**Comment by ignacio-chiazzo** [comment]
It was released on `v0.12.0`. Thanks for contributing.
